src/hooks/index.ts

```diff
@@ -245,6 +247,27 @@ export const useTokenInfo = (
   return query.data
 }
 
+export const useNftModerationStatus = ({wallet, fingerprint}: {wallet: YoroiWallet; fingerprint: string | null}) => {
+  return useQuery({
+    queryKey: [wallet.id, 'nft', fingerprint],
+    queryFn: () => (fingerprint ? wallet.fetchNftModerationStatus(fingerprint) : null),
+    enabled: !!fingerprint,
+  })
+}
+
+export const useTokenImage = ({wallet, tokenId}: {wallet: YoroiWallet; tokenId: string}): string | null => {
+  const {nfts} = useNfts(wallet)
+  const nft = nfts.find((nft) => nft.id === tokenId)
+  const fingerprint = nft ? getAssetFingerprint(nft.metadata.policyId, nft.metadata.assetNameHex) : null
+  const {data} = useNftModerationStatus({wallet, fingerprint})
+
+  if (nft && data === 'green') {
+    return nft.image
+  }
```

Comment:
maybe `useTokenImageModerated` and return an object `{image, status}` since the decision to open rely on the user when it is `yellow`. I'd also like to cache it later with the metadata + the moderate result.